src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -115,9 +122,27 @@ void HSHomeObject::init_homestore() {
     LOGI("Initialize and start HomeStore with app_mem_size = {}", homestore::in_bytes(app_mem_size));
 
     std::vector< homestore::dev_info > device_info;
-    for (auto const& path : app->devices()) {
-        device_info.emplace_back(std::filesystem::canonical(path).string(), homestore::HSDevType::Data);
+    bool has_data_dev = false;
+    bool has_fast_dev = false;
+    for (auto const& dev : app->devices()) {
+        auto input_dev_type = dev.type;
```

Comment:
We probably should obsolete this input.type (for now just ignore it?), for here we can just ignore input dev type and always take dev_type from iomgr as source of truth. It will avoid error case by a wrong input type is set from upper layer and conflicts iomgr type which cause later unexpected behaviors. Does it make sense?

Reply:
we on purpose want upper layer (finally , sysadmin , can override this, for specific testing case. Normally it should be set to auto_detect.

The UT is a clear use case for that,  our UT runs with files that locates in SSD,  we can simulate SSD/HDD hybrid case by providing input type.